x-pack/platform/plugins/shared/automatic_import/server/util/pipeline.ts

```diff
@@ -67,3 +68,59 @@ export async function createJSONInput(
   const { pipelineResults, errors } = await testPipeline(samples, pipeline, client);
   return { pipelineResults, errors };
 }
+
+/**
+ * Ensures that the given processors successfully compile.
+ *
+ * This works by simulating the pipeline in Elasticsearch and removes any processors that produce compile errors.
+ * In case we are unable to determine the failing processor, we recursively split the processors in half and try again.
+ *
+ * @param pipeline - The pipeline definition to simulate and validate.
+ * @param client - An Elasticsearch client scoped to the current user for pipeline simulation.
+ * @returns A promise resolving to the final pipeline definition after removing any failing processors.
+ */
+export async function ensureProcessorsCompile(
+  processors: ESProcessorItem[],
+  client: IScopedClusterClient
+): Promise<ESProcessorItem[]> {
+  const currentProcessors = [...processors];
+
+  while (currentProcessors.length > 0) {
+    const docs = [{ _source: { message: '' } }];
+    const pipeline = {
+      description: 'Pipeline for testing processor compilation',
+      processors: currentProcessors.map((p, index) => {
+        const key = Object.keys(p)[0];
+        return { [key]: { ...p[key], tag: `processor-${index}` } };
+      }),
+    };
+
+    try {
+      await client.asCurrentUser.ingest.simulate({ docs, pipeline });
+      break;
+    } catch (error) {
+      const tag = error.meta?.body?.error?.processor_tag;
+      if (tag) {
+        const index = toInteger(tag.split('-')[1]);
+        currentProcessors.splice(index, 1);
+      } else if (currentProcessors.length === 1) {
+        // There is only one processor left, and it is failing.
+        return [];
+      } else {
+        // Unable to determine the failing processor using the tag, we have to be creative.
```

Comment:
This logic is indeed a bit complex, and is not necessary if the Elasticsearch processor always returns the tag of offending processor as it should.